src/components/BulkEnrollmentPage/index.test.jsx

```diff
@@ -1,94 +1,138 @@
+    { status: 'assigned', count: 0 },
+    { status: 'activated', count: 2 }],
+});
 const subscriptions = Promise.resolve({
   data: {
-    results: {
-      subscriptions: [sub1, sub2],
-    },
+    results: [sub1, sub2],
     errors: null,
     setErrors: jest.fn(),
     forceRefresh: jest.fn(),
     loading: false,
+    count: 2,
   },
 });
-
-LicenseManagerApiService.fetchSubscriptions.mockReturnValue(subscriptions);
+const singleSubscription = Promise.resolve({
+  data: {
+    results: [sub1],
+    errors: null,
+    setErrors: jest.fn(),
+    forceRefresh: jest.fn(),
+    loading: false,
+    count: 1,
+  },
+});
+const mockSingleSubscriptionCount = Promise.resolve({
+  data: [
+    { status: 'unassigned', count: 0 },
+    { status: 'assigned', count: 0 },
+    { status: 'activated', count: 1 }],
+});
 
 const fakeStore = {
   portalConfiguration: {
     enterpriseId: 'foo',
+    enableCodeManagementScreen: false,
+    enableSubscriptionManagementScreen: true,
   },
 };
 const mockStore = configureMockStore([thunk]);
 const BulkEnrollmentWrapper = () => (
-  <MemoryRouter>
-    <Provider store={mockStore(fakeStore)}>
-      <BulkEnrollmentPage />
-    </Provider>
-  </MemoryRouter>
+  <Provider store={mockStore(fakeStore)}>
+    <BulkEnrollmentPage />
+  </Provider>
 );
 
-describe('<BulkEnrollmentPage />', () => {
+LicenseManagerApiService.fetchSubscriptions.mockImplementation(() => subscriptions);
+LicenseManagerApiService.fetchSubscriptionUsersOverview.mockImplementation(() => mockSubscriptionCount);
+
+describe('<BulkEnrollmentPage /> Multiple Subscriptions', () => {
   it('renders the bulk enrollment component', async () => {
     renderWithRouter(<BulkEnrollmentWrapper />);
     screen.getByText('Subscription Enrollment');
     await act(() => subscriptions);
   });
-  it.skip('renders the subscription picker', async () => {
-    renderWithRouter(<BulkEnrollmentWrapper />, { route: `/sluggo/admin/${ROUTE_NAMES.bulkEnrollment}` });
+  it('renders the subscription picker', async () => {
+    renderWithRouter(<BulkEnrollmentWrapper />, { route: `/${testSlug}/admin/${ROUTE_NAMES.bulkEnrollment}` });
     expect(await screen.findByText('Cohorts')).toBeInTheDocument();
     expect(await screen.findByText(sub1.title)).toBeInTheDocument();
     expect(await screen.findByText(sub2.title)).toBeInTheDocument();
-
+  });
+  it('renders the course search page', async () => {
+    renderWithRouter(<BulkEnrollmentWrapper />, { route: `/${testSlug}/admin/${ROUTE_NAMES.bulkEnrollment}/${sub1.uuid}` });
+    expect(await screen.findAllByText(ADD_COURSES_TITLE)).toHaveLength(2);
+  });
+  it('renders course search page when navigating via subscription picker', async () => {
+    renderWithRouter(<BulkEnrollmentWrapper />, { route: `/${testSlug}/admin/${ROUTE_NAMES.bulkEnrollment}` });
     await act(() => subscriptions);
+    // The expiration modal makes the enroll learners link inaccessible, so we need to dismiss it first.
```

Comment:
Should we make our test data so that there is no expiration modal? And make a separate test that tests that the expiration modal appears?

Reply:
So I did some fancy dynamic expiration date generation and then I realized that we just always show the expiration modal. Perhaps we should revisit that.... 😠 

```
return (
    <SubscriptionDetailContextProvider subscription={subscriptionFurthestFromExpiration}>
      {subscriptionFurthestFromExpiration.daysUntilExpiration > 0 && (
        <SubscriptionExpirationBanner />
      )}
```
and then:
```
  const renderMessage = () => (
    <>
      {hasMultipleSubscriptions && daysUntilExpiration <= 0 ? (
        <>
          This batch has expired. You are able to view the statuses of your invited learners.
        </>
      ) : (
        <>
          Your subscription is {daysUntilExpiration} days from expiration.
          Contact the edX Customer Success team at
          {' '}
          <MailtoLink to="customersuccess@edx.org">customersuccess@edx.org</MailtoLink>
          {' '}
          to extend your contract.
        </>
```
>_<

Reply:
this sounds like a new test ticket. Yeah ideally we avoid having that dialog altogether in tests, unless we are testing for it